client/ui/reducers/addonGroupsReducer.js

```diff
@@ -0,0 +1,199 @@
+      };
+
+    case FETCH_ADDON_GROUPS_SUCCESS:
+      return {
+        ...state,
+        list: {
+          addonGroups: action.addonGroups,
+          error: null,
+          loading: false,
+        },
+      };
+
+    case FETCH_ADDON_GROUPS_FAILURE:
+      return {
+        ...state,
+        list: {
+          addonGroups: [],
+          error: action.error,
+          loading: false,
+        },
+      };
+
+    case RESET_ADDON_GROUPS:
+      return {
+        ...state,
+        list: {
+          addonGroups: [],
+          error: null,
+          loading: false,
+        },
+      };
+
+    case CREATE_ADDON_GROUP_REQUEST:
+      return {
+        ...state,
+        create: {
+          addonGroup: null,
+          error: null,
+          loading: true,
+        },
+      };
+
+    case CREATE_ADDON_GROUP_SUCCESS:
+      return {
+        ...state,
+        create: {
+          addonGroup: action.addonGroup,
+          error: null,
+          loading: false,
+        },
+      };
+
+    case CREATE_ADDON_GROUP_FAILURE:
+      return {
+        ...state,
+        create: {
+          addonGroup: null,
+          error: action.error,
+          loading: false,
+        },
+      };
+
+    case RESET_NEW_ADDON_GROUP:
+      return {
+        ...state,
+        create: {
+          addonGroup: null,
+          error: null,
+          loading: false,
+        },
+      };
+
+    case FETCH_ADDON_GROUP_REQUEST:
+      return {
+        ...state,
+        active: {
+          addonGroup: null,
+          error: null,
+          loading: true,
+        },
+      };
+
+    case FETCH_ADDON_GROUP_SUCCESS:
+      return {
+        ...state,
+        active: {
+          addonGroup: action.addonGroup,
+          error: null,
+          loading: false,
+        },
+      };
+
+    case FETCH_ADDON_GROUP_FAILURE:
+      return {
+        ...state,
+        active: {
+          addonGroup: null,
+          error: action.error,
+          loading: false,
+        },
+      };
+
+    case RESET_ADDON_GROUP:
+      return {
+        ...state,
+        active: {
+          addonGroup: null,
+          error: null,
+          loading: false,
+        },
+      };
+
+    case UPDATE_ADDON_GROUP_REQUEST:
+      return {
+        ...state,
+        update: {
+          addonGroup: null,
+          error: null,
+          loading: true,
+        },
+      };
+
+    case UPDATE_ADDON_GROUP_SUCCESS:
+      return {
+        ...state,
+        update: {
+          addonGroup: action.addonGroup,
+          error: null,
+          loading: false,
+        },
+      };
+
+    case UPDATE_ADDON_GROUP_FAILURE:
+      return {
+        ...state,
+        update: {
+          addonGroup: null,
+          error: action.error,
+          loading: false,
+        },
+      };
+
+    case RESET_UPDATE_ADDON_GROUP:
+      return {
+        ...state,
+        update: {
+          addonGroup: null,
+          error: null,
+          loading: false,
+        },
+      };
```

Comment:
Yeah... I don't think we need to change it here, but I'm going to be on the look out for better patterns in the future. I'll let you know if I find anything.